sdk/maps/Azure.Maps.Search/src/MapsSearchClient.cs

```diff
@@ -892,19 +181,54 @@ public virtual async Task<Response<SearchAddressBatchResult>> GetImmediateSearch
+        ///
+        /// If query is given, should not use this parameter.
+        /// </param>
+        /// <param name="bbox">
+        /// A rectangular area on the earth defined as a bounding box object. The sides of the rectangles are defined by longitude and latitude values. When you specify this parameter, the geographical area is taken into account when computing the results of a location query.
+        ///
+        /// Example: lon1,lat1,lon2,lat2
+        /// </param>
+        /// <param name="view">
+        /// A string that represents an <see href="https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2">ISO 3166-1 Alpha-2 region/country code</see>. This will alter Geopolitical disputed borders and labels to align with the specified user region. By default, the View parameter is set to �Auto� even if you haven�t defined it in the request.
+        ///
+        /// Please refer to <see href="https://aka.ms/AzureMapsLocalizationViews">Supported Views</see> for details and to see the available Views.
+        /// </param>
+        /// <param name="coordinates"> A point on the earth specified as a longitude and latitude. When you specify this parameter, the user�s location is taken into account and the results returned may be more relevant to the user. Example: &amp;coordinates=lon,lat. </param>
+        /// <param name="adminDistrict">
+        /// The country subdivision portion of an address, such as WA. If query is given, should not use this parameter.
+        /// </param>
+        /// <param name="adminDistrict2">
+        /// The county for the structured address, such as King.
+        ///
+        /// If query is given, should not use this parameter.
+        /// </param>
+        /// <param name="adminDistrict3">
+        /// The named area for the structured address. If query is given, should not use this parameter.
+        /// </param>
+        /// <param name="locality">
+        /// The locality portion of an address, such as Seattle. If query is given, should not use this parameter.
+        /// </param>
+        /// <param name="postalCode">
+        /// The postal code portion of an address. If query is given, should not use this parameter.
+        /// </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual Response<SearchAddressBatchResult> GetImmediateSearchAddressBatch(IEnumerable<SearchAddressQuery> queries, CancellationToken cancellationToken = default)
+        public virtual Response<GeocodingResponse> GetGeocoding(string query = null, string addressLine = null, string countryRegion = null, IEnumerable<double> bbox = null, string view = null, IEnumerable<double> coordinates = null, string adminDistrict = null, string adminDistrict2 = null, string adminDistrict3 = null, string locality = null, string postalCode = null, int? top = null, CancellationToken cancellationToken = default)
```

Comment:
We need to wrap all the arguments to `GetGeocodingOptions`. [FuzzySearchOptions.cs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/maps/Azure.Maps.Search/src/Models/Options/FuzzySearchOptions.cs) is an example to reference, and we need to expand these options in this method by our own like how [FuzzySearch()](https://github.com/Azure/azure-sdk-for-net/blob/75c833536b595af7460ed6cd7ba4106f4f927296/sdk/maps/Azure.Maps.Search/src/MapsSearchClient.cs#L210) method does.

Reply:
Same for all other methods as long as there are more than 3 parameters to pass in.

Reply:
@dubiety done